src/company_cmd.cpp

```diff
@@ -1199,3 +1199,31 @@ CommandCost CmdGiveMoney(DoCommandFlag flags, uint32 money, CompanyID dest_compa
 	/* Subtract money from local-company */
 	return amount;
 }
+
+/**
+ * Get the index of the first available company. It attempts,
+ *  from first to last, and as soon as the attempt succeeds,
+ *  to get the index of the company:
+ *  1st - get the first existing human company.
+ *  2nd - get the first non-existing company.
+ *  3rd - get COMPANY_FIRST.
+ * @return the index of the first available company.
+ */
+CompanyID GetFirstAvailableCompany()
```

Comment:
I think the name for this function is wrong. After all, an AI is still an available company just not for what you want to try to do with this function. What about something like `GetCompanyToPutLocalPlayerInto`? That describes the intention of the function, making its uses clearer to the reader. How it's exactly to be implemented is something that can be explained in these comment.